File: crates/app/src/presentation/stash_entries.rs
use super_lazygit_core::state::StashItem;

use crate::presentation::icons::{icon_for_stash, is_icon_enabled};
use crate::style::basic_styles::fg_cyan;
use crate::style::text_style::TextStyle;
use crate::style::theme::{default_text_color, diff_terminal_color};

pub fn get_stash_entry_list_display_strings(
    stash_entries: &[StashItem],
    diff_name: &str,
) -> Vec<Vec<String>> {
    stash_entries
        .iter()
        .map(|stash_entry| {
            let diffed = stash_entry.stash_ref == diff_name;
            get_stash_entry_display_strings(stash_entry, diffed)
        })
        .collect()
}

fn get_stash_entry_display_strings(s: &StashItem, diffed: bool) -> Vec<String> {
    let text_style = if diffed {
        diff_terminal_color()
    } else {
        default_text_color()
    };

    let mut res = Vec::with_capacity(3);
    res.push(fg_cyan().sprint(&s.recency));

    if is_icon_enabled() {
        res.push(text_style.sprint(&icon_for_stash(s)));
    }

    res.push(text_style.sprint(&s.name));
    res
}

File: pkg/gui/presentation/stash_entries.go
package presentation

import (
	"github.com/quangdang46/slg/pkg/commands/models"
	"github.com/quangdang46/slg/pkg/gui/presentation/icons"
	"github.com/quangdang46/slg/pkg/gui/style"
	"github.com/quangdang46/slg/pkg/theme"
	"github.com/samber/lo"
)

func GetStashEntryListDisplayStrings(stashEntries []*models.StashEntry, diffName string) [][]string {
	return lo.Map(stashEntries, func(stashEntry *models.StashEntry, _ int) []string {
		diffed := stashEntry.RefName() == diffName
		return getStashEntryDisplayStrings(stashEntry, diffed)
	})
}

// getStashEntryDisplayStrings returns the display string of branch
func getStashEntryDisplayStrings(s *models.StashEntry, diffed bool) []string {
	textStyle := theme.DefaultTextColor
	if diffed {
		textStyle = theme.DiffTerminalColor
	}

	res := make([]string, 0, 3)
	res = append(res, style.FgCyan.Sprint(s.Recency))

	if icons.IsIconEnabled() {
		res = append(res, textStyle.Sprint(icons.IconForStash(s)))
	}

	res = append(res, textStyle.Sprint(s.Name))
	return res
}
